### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by knoxfighter**
Use github actions instead of travis. (most of the commits are for testing, i simply should have tested it locally somehow xD )

### CODE REVIEW

File: .github/workflows/create-release-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - uses: actions/setup-node@v1
+      - uses: actions/setup-go@v2
+      - uses: docker/setup-buildx-action@v1
+      - name: Get git tag
+        id: tag_name
+        uses: little-core-labs/get-git-tag@v3.0.2
+      - name: Login to dockerhub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
```
Comment (mroote) [inline .github/workflows/create-release-workflow.yml:51]: We need a service account for dockerhub uploads.

File: .github/workflows/test-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - '*.*'
+  pull_request:
+    branches:
+      - '**'
+env:
+  factorio_password: ${{ secrets.FACTORIO_PASSWORD }}
```
Comment (mroote) [inline .github/workflows/test-workflow.yml:11]: Can probably just make these generic strings instead of secrets for the tests.

Comment (knoxfighter) [reply .github/workflows/test-workflow.yml:11]: When you want your factorio credentials in a public githu repo, sure, do it. But i want to keep my username and password a secret :)

File: .github/workflows/create-release-workflow.yml (hunk 1 above)
Comment (knoxfighter) [reply .github/workflows/create-release-workflow.yml:51]: Currently i use my account (on my fork) and would also add it to this repo, i dont see the need of creating an additional dockerhub account.

Comment (mroote) [reply .github/workflows/create-release-workflow.yml:51]: Better to make a service account with limited access then to use your personal account for the uploads.

File: .github/workflows/test-workflow.yml (hunk 2 above)
Comment (mroote) [reply .github/workflows/test-workflow.yml:11]: Ah didn't realize valid credentials were needed for the tests.

**Pull request merged**